index.js

```diff
@@ -0,0 +1,128 @@
+
+document.addEventListener('mousemove', function (e) {
+    mouseX = e.pageX;
+    mouseY = e.pageY;
+
+    cursor.style.left = mouseX + 'px';
+    cursor.style.top = mouseY + 'px';
+});
+
+document.addEventListener('DOMContentLoaded', () => {
+    function initializePopup(images, trigger) {
+        const popup = document.getElementById('popup');
+        const closePopup = document.getElementById('closePopup');
+        const prev = document.getElementById('prev');
+        const next = document.getElementById('next');
+        const popupImage = document.getElementById('popupImage');
+
+        let currentIndex = 0;
+
+        function updatePopupImage(index) {
+            popupImage.src = images[index];
+        }
+
+        trigger.addEventListener('click', (e) => {
+            e.preventDefault();
+            popup.style.display = 'block';
+            updatePopupImage(currentIndex);
+        });
+
+        closePopup.addEventListener('click', () => {
+            popup.style.display = 'none';
+        });
+
+        window.addEventListener('click', (e) => {
+            if (e.target === popup) {
+                popup.style.display = 'none';
+            }
+        });
+
+        prev.addEventListener('click', () => {
+            currentIndex = (currentIndex === 0) ? images.length - 1 : currentIndex - 1;
+            updatePopupImage(currentIndex);
+        });
+
+        next.addEventListener('click', () => {
+            currentIndex = (currentIndex === images.length - 1) ? 0 : currentIndex + 1;
+            updatePopupImage(currentIndex);
+        });
+    }
+
+    const officeWebsiteTrigger = document.getElementById('officeWebsiteTrigger');
+    initializePopup(["./img/picture2.png", "./img/picture3.png", "./img/picture1.png"], officeWebsiteTrigger);
+
+    const formGeneratorTrigger = document.getElementById('formGeneratorTrigger');
+    initializePopup(["./img/picture1.png", "./img/picture2.png", "./img/picture3.png"], formGeneratorTrigger);
+});
+
+function addStylesToActiveLinkOnScroll() {
+    var links = document.querySelectorAll('a');
+
+    function getCurrentSection() {
+        var currentScroll = window.scrollY;
+        var sections = document.querySelectorAll('section');
+        var currentSectionId = '';
+
+        sections.forEach(function (section) {
+            var sectionTop = section.offsetTop - 50;
+            var sectionBottom = sectionTop + section.offsetHeight;
+
+            if (currentScroll >= sectionTop && currentScroll < sectionBottom) {
+                currentSectionId = '#' + section.getAttribute('id');
+            }
+        });
+
+        var documentHeight = document.documentElement.scrollHeight;
+        var windowHeight = window.innerHeight;
+        if (currentScroll + windowHeight >= documentHeight) {
+            currentSectionId = '#' + sections[sections.length - 1].getAttribute('id');
+        }
+
+        return currentSectionId;
+    }
+
+    function setActiveLinkStyle() {
```

Comment:
zamien to na ES6  i kazda inna funkcje tez 



const setActiveLinkStyle = () => {
    const currentSection = getCurrentSection();

    links.forEach(link => {
        link.style.fontSize = '';
        link.style.color = '';

        if (link.getAttribute('href') === currentSection) {
            link.style.color = '#99ccff';
        }
    });
};